### PR TITLE
Feature/search vm

### PR DESCRIPTION
**Pull request by nikilark**
Search service added.
You can search only when loged in (only on pages with navbar, not logInNavbar). To search type something in searchbar and press enter. You can search by name, surname and username separately or together. You can type several users through come. There should be no repeating users in result. If you type something like numbers or special characters, there would be no results. 
Examples :
1) Oleg, Roman, Petro -- three results
2) Oleg, Roman Levkovych, RomanLevkovych -- two results
3) 12345 -- nothing was found
4) 12345, Roman -- returns Roman

### CODE REVIEW

File: DataServices/DataServiceBase.cs [hunk 1]
```diff
@@ -128,6 +146,11 @@ protected virtual void Dispose(bool disposing)
                 }
             }
         }
+
+        string IFaqService.LoadAnswerForQuestionAsync(string question)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (RomanLevkovych) [inline DataServices/DataServiceBase.cs]: Why this method is not implemented? 
Remove if it is not needed and move out from `Disposable` region.

File: View/Pages/FaqPageLoged.xaml.cs [hunk 2]
```diff
@@ -28,6 +28,16 @@ public FaqPageLoged()
             navbar.button_Notes.Click += Navbar_Button_Notes_Click;
             navbar.button_TimeTable.Click += Navbar_Button_TimeTable_Click;
             navbar.button_FAQ.Style = Application.Current.Resources["MenuButtonActive"] as Style;
+            searchBar.textSearch.KeyDown += SearchBar_TextSearch_KeyDown;
+        }
+
+        private void SearchBar_TextSearch_KeyDown(object sender, KeyEventArgs e)
```
Comment (RomanLevkovych) [inline View/Pages/FaqPageLoged.xaml.cs]: what about meaningful name?

File: View/Pages/NotesPage.xaml.cs [hunk 3]
```diff
@@ -44,6 +44,16 @@ public NotesPage()
             content.button_sortSubject.Click += Content_Button_sortCreated_Click;
             content.button_sortName.Click += Content_Button_sortDeadline_Click;
             SortNotes(sorted);
+            searchBar.textSearch.KeyDown += SearchBar_TextSearch_KeyDown;
+        }
+
+        private void SearchBar_TextSearch_KeyDown(object sender, KeyEventArgs e)
```
Comment (RomanLevkovych) [inline View/Pages/NotesPage.xaml.cs]: The same as above

File: View/Pages/ProfilePage.xaml.cs [hunk 4]
```diff
@@ -28,6 +28,16 @@ public ProfilePage()
             navbar.button_Notes.Click += Navbar_Button_Notes_Click;
             navbar.button_TimeTable.Click += Navbar_Button_TimeTable_Click;
             navbar.button_Profile.Style = Application.Current.Resources["MenuButtonActive"] as Style;
+            searchBar.textSearch.KeyDown += SearchBar_TextSearch_KeyDown;
+        }
+
+        private void SearchBar_TextSearch_KeyDown(object sender, KeyEventArgs e)
```
Comment (RomanLevkovych) [inline View/Pages/ProfilePage.xaml.cs]: same as above

File: DataServices/DataServiceBase.cs (hunk 1 above)
Comment (Scaler358) [reply DataServices/DataServiceBase.cs]: Cause Im not doing FAQ page

Comment (RomanLevkovych) [reply DataServices/DataServiceBase.cs]: There already are this method implemented, make it public and remove this one.

**Review by arihilmir — APPROVED** [review]

**Pull request merged**